src/node_modules/_file-explorer.js

```diff
@@ -0,0 +1,292 @@
+var setval = require('setval')
+var yo = require('yo-yo')
+var csjs = require('csjs-inject')
+var Treeview = require('ethereum-remix').ui.TreeView
+
+var css = csjs`
+  .treeview {
+    display: block;
+    margin-top: 7px;
+    width: 20%;
+  }
+  .folder,
+  .file {
+    font-size: 14px;
+  }
+  .hasFocus {
+    background-color: #F4F6FF;
+  }
+  .rename {
+    background-color: white;
+  }
+  .remove,
+  .save {
+    align-self: center;
+    margin-right: 3%;
+  }
+  .activeMode {
+    display: flex;
+    justify-content: space-between;
+  }
+`
+module.exports = fileExplorer
+
+function fileExplorer (container, files, editor) {
```

Comment:
so the `api` would have certain mainly "read" events, like `'list'` and certain more "write" events, like `'remove'`, even though removing might still return a boolean, thus the callback `cb`.
**The above would be for communication initiated by `fileExplorer`**

**For communication initiated by `app.js` maybe something like the following would work:**
```js
// app.js
files.event.register('fileRenamed', function (oldName, newName) {
  api.emit('fileRenamed', [oldName, newName], function response () { })
})
```
this pattern could inform the `fileExplorer` instance about stuff.
While `fileRenamed` does not need any response data, some other commands custom to the file-explorer might need a response - at least the response could inform about error or success